GettingStarted/src/Samples/Positioning/SGSPositioningVC.m

```diff
@@ -0,0 +1,143 @@
+//
+//  SGSPositioningVC.m
+//  GettingStarted
+//
+//  Created by Fernando Sanchez Vilas on 28/03/2019.
+//  Copyright © 2019 Situm Technologies S.L. All rights reserved.
+//
+
+#import "SGSPositioningVC.h"
+@import CoreLocation;
+
+@interface SGSPositioningVC () <SITLocationDelegate>
+
+@property (weak, nonatomic) IBOutlet UIButton *startPositioningButton;
+@property (weak, nonatomic) IBOutlet UILabel *statusLabel;
+@property (weak, nonatomic) IBOutlet UILabel *positionLabel;
+
+@property (nonatomic, strong) CLLocationManager *locationManager;
+@property (nonatomic) BOOL isPositioning;
+@property (nonatomic, strong) NSDictionary *locationStatesTranslator;
+
+@end
+
+// Request user permission strings
+static NSString *PermissionDeniedAlertTitle = @"Location Authorization Needed";
+static NSString *PermissionDeniedAlertBody = @"This app needs location authorization to work properly. Please go to settings and enable it.";
+static NSString *PermissionDeniedOk = @"Ok";
+
+@implementation SGSPositioningVC
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self customizeNavigationBarWithTitle:@"Positioning"];
+    self.isPositioning = NO;
+    self.locationStatesTranslator = @{@(0):@"Stopped",
+                                  @(1):@"Calculating",
+                                  @(2):@"Not In Building",
+                                  @(3):@"Started",
+                                  @(4):@"Needs Calibration"
+                                  };
+    self.locationManager = [[CLLocationManager alloc] init];
+    [SITLocationManager sharedInstance].delegate = self;
+}
+
+-(void)viewDidAppear:(BOOL)animated{
+    [super viewDidAppear:animated];
+    //Has to be done after the view has appeared otherwise the alert wont last enough
+    [self requestLocationAuthorization];
+}
+
+-(void)viewWillDisappear:(BOOL)animated{
+    [super viewWillDisappear:animated];
+    [self stopPositioning];
+}
+
+#pragma mark -- Request Location Authorization
+-(void)requestLocationAuthorization{
```

Comment:
Esta lóxica probablemente se empregue en máis exemplos. Crees que sería factible (e que merece a pena) pasalo o `SGSSamplesBaseVC`?

Reply:
Realmente esto non se emprega aquí, xa que pasa antes pola da pantalla de selección de exemplo, onde xa se chequea.
Engadino a propósito para que o exemplo fose autocontido e contivese tódolos pasos na clase sen ter que ir buscar cousas noutro lado. 
Algo similar ó que comentas, pasoume coa propiedade @property(nonatomic,strong) SITBuildingInfo *selectedBuildingInfo; que inicialmente meteraa en SGSSamplesBaseVC, pero despois de darlle unha volta paseina para cada un dos VC dos exemplos, desta forma non tes que andar mirando onde está declarada.
A decisión que tomei foi meter en SGSSamplesBaseVC, detalles de navegación, etc que realmente a quen está mirando o exemplo non lle preocupan pero non cousas do SDK de Situm.
Inda que podo cambiar a estratexia se queredes

Reply:
Non é necesario, convenceme bastante a separación de responsabilidades como a definiches. Por mi pode quedar asi 👍